GitUI/UserControls/RevisionGrid/Graph/RevisionGraph.cs

```diff
@@ -533,35 +633,46 @@ private RevisionGraphRevision[] BuildOrderedNodesCache(int currentRowIndex)
             {
                 try
                 {
-                    // Reset the reorder flag and the orderedUntilScore. This makes sure it isn't marked dirty before we even got to
-                    // rebuilding it.
+                    // Reset flags to sure it isn't marked dirty before we even got to rebuilding it.
                     _orderedUntilScore = int.MinValue;
-                    _reorder = false;
+                    _reorderNodeCache = false;
+
+                    RevisionGraphRevision[] nodes = _revisionByObjectId.Values.ToArray();
+                    Array.Sort(nodes, (a, b) => a.Score.CompareTo(b.Score));
+                    _orderedNodesCache = nodes.ToImmutableArray();
 
-                    // Use a local variable, because the cached list can be reset.
-                    RevisionGraphRevision[] localOrderedNodesCache = _nodes.ToArray();
-                    Array.Sort(localOrderedNodesCache, (x, y) => x.Score.CompareTo(y.Score));
-                    _orderedNodesCache = localOrderedNodesCache;
-                    if (localOrderedNodesCache.Length > 0)
+                    if (_orderedNodesCache.Length > 0)
                     {
-                        _orderedUntilScore = localOrderedNodesCache[^1].Score;
+                        _orderedUntilScore = _orderedNodesCache[^1].Score;
                     }
 
-                    return localOrderedNodesCache;
+                    return _orderedNodesCache;
```

Comment:
Are you really sure that the graph cannot be cleared in parallel.
`BuildOrderedNodesCache(currentRowIndex)` promises to return at least `currentRowIndex` rows.
The previous implementation is safer at no cost.

Reply:
I believe BuildOrderedNodesCache() could be called in parallel, but one of the calls will win. There is a theoretical risk that _orderedNodesCache is set from the incorrect call though.
Removing _orderedUntilScore .
a local copy of _orderedNodesCache is cheap

Also changed the sorting, as I got an exception in some tests, the code used is faster too (but probably allocating  more).